src/main/scala/com/fulcrumgenomics/vcf/filtration/ReadEndSomaticVariantFilter.scala

```diff
@@ -24,49 +24,49 @@
 
 package com.fulcrumgenomics.vcf.filtration
 
-import java.lang.Math.{min, pow}
-
 import com.fulcrumgenomics.bam.{Bams, BaseEntry, Pileup, PileupEntry}
 import com.fulcrumgenomics.commons.util.LazyLogging
 import com.fulcrumgenomics.util.NumericTypes.{LogProbability => LnProb}
 import com.fulcrumgenomics.util.Sequences
 import com.fulcrumgenomics.vcf.api.{VcfCount, VcfFieldType, VcfInfoHeader, _}
 
-/**
-  * Filter that examines SNVs with the alt allele being A or T to see if they are likely the result
-  * of aberrant end-repair and A-base addition.
-  *
-  * Occurs when end-repair is performed in a way that chews back single-stranded 3' overhangs
-  * to create a blunt end, but over-digests and ends up creating a recessed 3' end which subsequently
-  * (and incorrectly) gets filled in with one or more As during the A-base addition step. The result
-  * is one or more base substitutions to A at the 3' end of molecules, which after PCR also shows up
-  * as substitutions to T at the 5' end.
-  *
-  * @param distance The distance from the end of the template that defines the region where this
-  *                 artifact may occur
-  * @param pValueThreshold the pvalue at and below which mutations are filtered as being likely
-  *                        caused by the artifact
-  */
-class EndRepairArtifactLikelihoodFilter(val distance: Int = 2,
-                                        pValueThreshold: Option[Double] = None) extends SomaticVariantFilter with LazyLogging {
+import java.lang.Math.{min, pow}
 
-  /** The INFO header line needed by this filter. */
-  val Info = VcfInfoHeader("ERAP", VcfCount.Fixed(1), VcfFieldType.Float, "P-Value for the End-Repair Artifact test.")
+/** Trait for classes that apply filters to somatic variants calls at read ends. */
+trait ReadEndSomaticVariantFilter extends SomaticVariantFilter with LazyLogging {
 
-  /** The FILTER header line needed by this filter. */
-  val Filter = VcfFilterHeader("EndRepairArtifact", "Variant is likely an artifact caused by end-repair and A-base addition.")
+  /** The VCF header line that describes the INFO key that the filter may reference. */
+  def Info: VcfInfoHeader
```

Comment:
How about `def infoHeadLine` to make it clear in usage below that it's the entire header line and also lower case first letter to indicate def.  Or make it `val InfoHeaderLine`?